src/Core/src/Result.php

```diff
@@ -62,13 +62,41 @@ public function __destruct()
         return $this->response->resolve($timeout);
     }
 
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $results
+     * @param float|null $timeout      Duration in seconds before aborting. When null wait
+     *                                 until the end of execution. Using 0 means non-blocking
+     * @param bool       $downloadBody Wait until receiving the entire response body or only the first bytes
+     *
+     * @return iterable<self>
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function multiplex(iterable $results, float $timeout = null, bool $downloadBody = false): iterable
+    {
+        $resultMap = [];
+        $responses = [];
+        foreach ($results as $index => $result) {
+            $responses[$index] = $result->response;
+            $resultMap[\spl_object_id($result->response)] = $result;
```

Comment:
We dont need `\spl_object_id($result->response)` here. We can use use `$index`.

Reply:
good catch!!